### PR TITLE
Nov GA and preview changes for analyze-text api

### LINKED ISSUE #31012: [Cognitive Services Text Analytics API ] API Review
New API Review meeting has been requested.

**Service Name**: Cognitive Services Text Analytics API 
**Review Created By**: Aurgho Bhattacharjee
**Review Date**: 10/16/2024 09:00 AM PT
**Release Plan**: [](https://web.powerapps.com/apps/821ab569-ae60-420d-8264-d7b5d5ef734c?release-plan-id=)
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/30966
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: GA and preview api for analyze-text.
NER: Metadata, tags type to be GAd without breaking customers.
PII: promote exclusion list in text pii to GA. Not promoting redaction character.
Language Detection : promote script code feature to GA. 
Summarization : Change query parameter to prompt in abstractive summarization. GA sumary length parameter.
Healthcare: GA of temporal assertions and FHIR. 
Deprecation of custom healthcare, custom sentiment, custom abstractive summarization.

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=bf24ee87-608a-ef11-ac21-6045bd019375)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by aurghob**
# Data Plane API Specification Update Pull Request

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## API Info: The Basics

Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [x] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:
Adding new API 2024-11-01 and 2024-11-15-preview for analyze-text.
1. GA EntityRecognition tags and type in output. Preseve previous properties to avoid breaking changes for NER.
2. GA EntityRecognition input parameters inclusion and exclusion list to filter tags.
3. GA SummaryLength input parameter for abstractive summarization.
4. GA Healthcare FHIR and Healthcare temporality assertion.
5. Preview Tags and Type in output in PIIEntityRecognition.
6. Preview redaction policy in input parameters for PIIEntityRecognition.

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories) 
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

fix https://github.com/Azure/azure-rest-api-specs/issues/31012

### CODE REVIEW

File: specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp [hunk 1]
```diff
@@ -32,12 +32,63 @@ model PiiTaskParameters is PreBuiltTaskParameters {
   stringIndexType?: StringIndexType = StringIndexType.TextElements_v8;
 
   /** Enumeration of PII categories to be excluded in the response. */
-  @added(Versions.v2023_11_15_preview)
+  @added(Versions.v2024_11_15_preview)
   excludePiiCategories?: Array<PiiCategoriesExclude>;
 
   /** Character to be used for redaction. Default in '*'. */
-  @added(Versions.v2023_11_15_preview)
+  @added(Versions.v2024_11_15_preview)
```
Comment (bidisha-c) [inline specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp:39]: We need to remove this from the nov preview

File: specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp [hunk 2]
```diff
@@ -32,12 +32,63 @@ model PiiTaskParameters is PreBuiltTaskParameters {
...
+  /** The entity RedactionPolicy object kind. */
+  policyKind: RedactionPolicyKind.characterMask;
+
+  /** Optional parameter to use a Custom Character to be used for redaction in PII responses. Default character will be * as before. We allow specific ascii characters for redaction. */
+  redactionCharacter?: redactionCharacter = redactionCharacter.asterisk;
+}
+
+/** Represents the policy of not redacting found PII. */
+@added(Versions.v2024_11_15_preview)
+model DoNotRedactPolicyType extends BaseRedactionPolicy {
```
Comment (bidisha-c) [inline specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp:67]: NoMaskPolicyType

File: specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp [hunk 3]
```diff
@@ -32,12 +32,63 @@ model PiiTaskParameters is PreBuiltTaskParameters {
...
+@added(Versions.v2024_11_15_preview)
+@discriminator("policyKind")
+model BaseRedactionPolicy {
+  /** The entity RedactionPolicy object kind. */
+  policyKind: RedactionPolicyKind = RedactionPolicyKind.characterMask;
+}
+
+/** Represents the policy of redacting with a redaction character */
+@added(Versions.v2024_11_15_preview)
+model MaskWithRedactionCharacterPolicyType extends BaseRedactionPolicy {
```
Comment (bidisha-c) [inline specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp:57]: CharacterMaskPolicyType

File: specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp [hunk 4]
```diff
@@ -32,12 +32,63 @@ model PiiTaskParameters is PreBuiltTaskParameters {
...
+  @added(Versions.v2024_11_15_preview)
+  redactionPolicy?: BaseRedactionPolicy;
+}
+
+/** The abstract base class for RedactionPolicy. */
+@added(Versions.v2024_11_15_preview)
+@discriminator("policyKind")
+model BaseRedactionPolicy {
+  /** The entity RedactionPolicy object kind. */
+  policyKind: RedactionPolicyKind = RedactionPolicyKind.characterMask;
```
Comment (bidisha-c) [inline specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp:52]: Is this how we do defaults?

File: specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp [hunk 5]
```diff
@@ -32,12 +32,63 @@ model PiiTaskParameters is PreBuiltTaskParameters {
...
+}
+
+/** Represents the policy of redacting with a redaction character */
+@added(Versions.v2024_11_15_preview)
+model MaskWithRedactionCharacterPolicyType extends BaseRedactionPolicy {
+  /** The entity RedactionPolicy object kind. */
+  policyKind: RedactionPolicyKind.characterMask;
+
+  /** Optional parameter to use a Custom Character to be used for redaction in PII responses. Default character will be * as before. We allow specific ascii characters for redaction. */
+  redactionCharacter?: redactionCharacter = redactionCharacter.asterisk;
```
Comment (bidisha-c) [inline specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp:62]: do we need to mention that in typespec as redactionCharacter is an optional parameter

File: specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp (hunk 1 above)
Comment (aurghob) [reply specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp:39]: removing

File: specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp (hunk 4 above)
Comment (aurghob) [reply specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp:52]: yes.

File: specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp (hunk 5 above)
Comment (aurghob) [reply specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp:62]: Following the pattern of our optional parameters and keeping it for now.

File: specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp (hunk 3 above)
Comment (aurghob) [reply specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp:57]: Missed this cleanup in last commit - added.

File: specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp (hunk 4 above)
Comment (aurghob) [reply specification/cognitiveservices/Language.AnalyzeText/models/pii.entity.recognition.tsp:48]: keeping

**Review by mikekistler — APPROVED** [review]
Looks good. 👍

**Pull request merged**